spring-cloud-sleuth-core/src/main/java/org/springframework/cloud/sleuth/instrument/web/client/feign/TraceFeignObjectWrapper.java

```diff
@@ -57,12 +60,31 @@ Object wrap(Object bean) {
 		if (bean instanceof Client && !(bean instanceof TracingFeignClient)) {
 			if (ribbonPresent && bean instanceof LoadBalancerFeignClient
 					&& !(bean instanceof TraceLoadBalancerFeignClient)) {
-				LoadBalancerFeignClient client = ((LoadBalancerFeignClient) bean);
-				return new TraceLoadBalancerFeignClient(
-						(Client) new TraceFeignObjectWrapper(this.beanFactory)
-								.wrap(client.getDelegate()),
-						factory(), (SpringClientFactory) clientFactory(),
-						this.beanFactory);
+				if (AopUtils.getTargetClass(bean).equals(LoadBalancerFeignClient.class)) {
+					LoadBalancerFeignClient client = ((LoadBalancerFeignClient) bean);
+					return new TraceLoadBalancerFeignClient(
+							(Client) new TraceFeignObjectWrapper(this.beanFactory)
+									.wrap(client.getDelegate()),
+							factory(), (SpringClientFactory) clientFactory(),
+							this.beanFactory);
+				}
+				else {
+					LoadBalancerFeignClient client = ((LoadBalancerFeignClient) bean);
+					try {
+						Field delegate = LoadBalancerFeignClient.class.getDeclaredField("delegate");
+						delegate.setAccessible(true);
+						delegate.set(client, new TraceFeignObjectWrapper(this.beanFactory).wrap(client.getDelegate()));
+					}
+					catch (NoSuchFieldException | IllegalArgumentException | IllegalAccessException | SecurityException e) {
+						// not sure the best option here
```

Comment:
I think that ignoring this is a good idea. We can try to just return a wrapped instance or return a non wrapped one with a warning message. I think that we shouldn't break the application startup due to tracing. In case of any exception I could log a warning message and fallback to only wrapping the delegate as you suggested